tests/integration/workflows/test_workflow_core_selection.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+from __future__ import absolute_import, division, print_function
+
+import json
+import mock as mock
+
+import os
+
+import pkg_resources
+import requests_mock
+from flask import current_app
+from invenio_workflows import start, workflow_object_class, ObjectStatus, WorkflowObject
+from invenio_workflows.models import WorkflowObjectModel
+from mocks import fake_beard_api_request, fake_magpie_api_request
+from utils import override_config
+
+
+def load_json_record(record_file):
+    return json.loads(pkg_resources.resource_string(
+        __name__,
+        os.path.join(
+            'fixtures',
+            record_file
+        )
+    ))
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.submit_rt_ticket', return_value="1234")
+@mock.patch(
+    "inspirehep.modules.workflows.tasks.beard.json_api_request",
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    "inspirehep.modules.workflows.tasks.magpie.json_api_request",
+    side_effect=fake_magpie_api_request,
+)
+def test_core_selection_wf_starts_after_article_wf_when_no_core(mocked_api_request_magpie, mocked_api_request_beard, mocked_rt, workflow_app, mocked_external_services):
+    pid_value = 123456
+    mocked_url = "{inspirehep_url}/{endpoint}/{control_number}".format(
+        inspirehep_url=current_app.config.get("INSPIREHEP_URL"),
+        endpoint='literature',
+        control_number=pid_value
+    )
+    record = {
+        "_collections": [
+            "Literature"
+        ],
+        "titles": [
+            {"title": "A title"},
+        ],
+        "document_type": [
+            "report"
+        ],
+        "collaborations": [
+            {"value": "SHIP"}
+        ],
+        "control_number": pid_value,
+    }
+
+    workflow_object = workflow_object_class.create(
+        data=record,
+        id_user=None,
+        data_type='hep'
+    )
+    workflow_object.extra_data['source_data'] = {"data": record, "extra_data": {"source_data": {"data": record}}}
+    workflow_object.extra_data['approved'] = 'True'
+    workflow_object.save()
+
+    with override_config(FEATURE_FLAG_ENABLE_REST_RECORD_MANAGEMENT=True):
+        with requests_mock.Mocker() as mock:
+            mock.register_uri('GET', mocked_url, json=load_json_record('hep_record_no_core.json'))
+            # It's update so it should be PUT
+            mock.register_uri('PUT', "http://web:8000/literature/{control_number}".format(control_number=pid_value), json={"metadata": {"control_number": pid_value}})
+
+            start("article", object_id=workflow_object.id)
+
+            assert WorkflowObjectModel.query.filter(WorkflowObjectModel.workflow.has(name="core_selection")).count() == 0
+
+            workflow_object.callback_pos = [34, 1, 13]
```

Comment:
seems very brittle, isn't there a better way?

Reply:
There is always a better way but making a WF in tests to go through whole process and finish is not easy and creating test like that will take time :/ Still it's not required for WF to really go through as we do not care for it really in this test We just want to test if it's creating another workflow at the end if record is not core.

Reply:
I'm worried about maintainability of this when the workflow changes. But maybe it's too late for maintainability of workflow tests anyway :)